src/materialized/src/http/prof.rs

```diff
@@ -59,8 +59,24 @@ struct FlamegraphTemplate<'a> {
 async fn time_prof<'a>(
     params: &HashMap<Cow<'a, str>, Cow<'a, str>>,
 ) -> anyhow::Result<Response<Body>> {
+    let _ctl_lock;
+    cfg_if! {
+        if #[cfg(target_os = "macos")] {
+            _ctl_lock = ();
+        } else {
+            _ctl_lock = if let Some(ctl) = prof_jemalloc::PROF_CTL.as_ref() {
+                let mut borrow = ctl.lock().await;
+                borrow.deactivate()?;
+                Some(borrow)
+            } else {
+                None
+            };
+        }
+    }
     let merge_threads = params.get("threads").map(AsRef::as_ref) == Some("merge");
-    let stacks = prof_time(Duration::from_secs(10), 99, merge_threads).await?;
+    // SAFETY: We ensure above that memory profiling is off.
+    // Since we hold the mutex, nobody else can be turning it back on in the intervening time.
+    let stacks = unsafe { prof_time(Duration::from_secs(10), 99, merge_threads) }.await?;
```

Comment:
it might be worth adding an explicit `drop(_ctl_lock);` after this to ensure that we maintain ownership of the lock in the face of refactorings.

Reply:
Done